docs2/site/docs/guides/migration3.md

```diff
@@ -0,0 +1,430 @@
+    var result = await executer.ExecuteAsync(options =>
+    {
+        options.Schema = schema;
+        options.Query = request.Query;
+        options.OperationName = request.OperationName;
+        options.Inputs = request.Variables.ToInputs();
+    });
+
+    context.Response.ContentType = "application/json";
+    context.Response.StatusCode = 200; // OK
+
+    var writer = new GraphQL.SystemTextJson.DocumentWriter();
+    await writer.WriteAsync(context.Response.Body, result);
+}
+
+public class GraphQLRequest
+{
+    public string OperationName { get; set; }
+
+    public string Query { get; set; }
+
+    [JsonConverter(typeof(GraphQL.SystemTextJson.ObjectDictionaryConverter))]
+    public Dictionary<string, object> Variables { get; set; }
+}
+```
+
+This is just a simplified example of the changes necessary. Note that typically the `DocumentExecuter` and `DocumentWriter`
+are registered as singletons within the dependency injection container, as they can safely be shared between requests.
+
+If you continue to use the `Newtonsoft.Json` converter, please note that Asp.Net Core 3.0 disallows synchronous IO by default,
+which is required by the converter.  You will need to make a change in the `ConfigureServices` section of `Startup.cs` as
+follows:
+
+```csharp
+// kestrel
+services.Configure<KestrelServerOptions>(options =>
+{
+    options.AllowSynchronousIO = true;
+});
+
+// IIS
+services.Configure<IISServerOptions>(options =>
+{
+    options.AllowSynchronousIO = true;
+});
+```
+
+Please note that if you use a `NameConverter` other than the default `CamelCaseNameConverter`, you may need to configure
+your json serializer also to not convert object properties to camel case. For example, with the `System.Text.Json` converter,
+you need to set `JsonSerializerOptions.PropertyNamingPolicy = null;` as follows:
+
+```csharp
+var writer = new GraphQL.SystemTextJson.DocumentWriter(options => {
+    options.PropertyNamingPolicy = null;
+});
+```
+
+### UserContext
+
+The definition of the `UserContext` object throughout the library has changed to a `IDictionary<string, object>`.
+Custom user context classes will need to inherit from `Dictionary<string, object>` or otherwise support the interface.
+
+```csharp
+//class MyContext
+class MyContext : Dictionary<string, object>
+{
+    public DbContext MyDbContext { get; set; }
+}
+```
+
+### Document Listeners
+
+The `DocumentExecutionListenerBase<T>` class and `IDocumentExecutionListener<T>` interface have been removed;
+please implement the `IDocumentExecutionListener` interface when creating a custom document listener. You
+can also inherit from the `DocumentExecutionListenerBase` class to provide default implementations of events.
+
+The methods definitions have also changed from passing `userContext` and `cancellationToken` parameters to a
+single parameter `context` of type `IExecutionContext`.  The context has properties for accessing the user context,
```

Comment:
```suggestion
single parameter `context` of type `IExecutionContext`. The context has properties for accessing the user context,
```

Reply:
So dumb that html doesn't support two spaces.  Not that there's anything that can be done about it considering the design of html.  It just aggravates me to no end to see improperly spaced paragraphs.